### PR TITLE
Fix Context creation/handling for multikeys

### LINKED ISSUE #135: MultiKeys require exposing multiple contexts
Currently Flow sets up and delivers a single context for an incoming key. That's insufficient for multikeys, where we should deliver a context for each component key (plus the multikey itself).

### PR DESCRIPTION
**Pull request by loganj**
Fixes #135 

Hard part was already done, forgot I saw this coming.  Most real changes are in KeyDispatcher, which has to actually create the Contexts for the KeyChanger to use.

@rjrjr

### CODE REVIEW

File: flow/src/main/java/flow/KeyDispatcher.java [hunk 1]
```diff
@@ -65,14 +69,30 @@ private KeyDispatcher(Activity activity, KeyChanger keyChanger) {
     // TODO(#126): this short-circuit may belong in Flow, since every Dispatcher we have implements it.
     if (inKey.equals(outKey)) {
```
Comment (rjrjr) [inline flow/src/main/java/flow/KeyDispatcher.java]: Up above, you still have @Nullable on the KeyChanger field.

File: flow/src/main/java/flow/KeyDispatcher.java [hunk 2]
```diff
@@ -65,14 +69,30 @@ private KeyDispatcher(Activity activity, KeyChanger keyChanger) {
     // TODO(#126): this short-circuit may belong in Flow, since every Dispatcher we have implements it.
```
Comment (rjrjr) [inline flow/src/main/java/flow/KeyDispatcher.java]: If you build the short circuit into flow, then you can also build the rest of this method into flow, and KeyDispatcher/KeyChanger again stop looking necessary.

**Comment by rjrjr** [comment]
I made comments but we can address in follow up.

File: flow/src/main/java/flow/KeyDispatcher.java (hunk 2 above)
Comment (loganj) [inline flow/src/main/java/flow/KeyDispatcher.java]: Stop looking necessary in that KeyChanger would effectively replace Dispatcher, right?

**Pull request merged**